### PR TITLE
add checkstyle

### PR DESCRIPTION
**Pull request by casz**
### Your checklist for this pull request

🚨 Please review the [guidelines for contributing](../blob/master/docs/CONTRIBUTING.md) to this repository.

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side) and not your master branch!

- [x] Please describe what you did

- [ ] Link to relevant GitHub issues or pull requests

- [ ] Link to relevant [Jenkins JIRA issues](https://issues.jenkins-ci.org)

- [ ] Did you provide a test-case? That demonstrates feature works or fixes the issue.

### Description

A sane checkstyle and import order

### CODE REVIEW

File: plugin/config/suppressions.xml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0"?>
+
+<!DOCTYPE suppressions PUBLIC
+  "-//Checkstyle//DTD SuppressionFilter Configuration 1.2//EN"
+  "https://checkstyle.org/dtds/suppressions_1_2.dtd">
+
+<suppressions>
+  <suppress checks="AvoidStarImport" files=".*Test\.java"/>
```
Comment (timja) [inline plugin/config/suppressions.xml]: not really sure this exclusion gives much value?

Comment (casz) [reply plugin/config/suppressions.xml]: star imports in test files I don't mind but if you disagree let's avoid start import everywhere 😅

Comment (casz) [reply plugin/config/suppressions.xml]: what do you think about applying it on all modules except snakeyaml?

Comment (timja) [reply plugin/config/suppressions.xml]: I would prefer one rule everywhere, simpler to reason with and not sure if you could configure the IDE to know in test files its fine to star import

Comment (timja) [reply plugin/config/suppressions.xml]: what's the reason for not doing it in snakeyaml?
fine with it if it would be annoying there

Comment (casz) [reply plugin/config/suppressions.xml]: You cannot I had to edit my IDE config to optimize once for tests and once for main 😰

Comment (casz) [reply plugin/config/suppressions.xml]: We have snakeyaml included so we can shade it and makes no sense to maintain the code.

I would like if possible to get RID of snakeyaml and shade it differently.
@oleg-nenashev I can't remember, but you had some suggestion on how to shade it.

Comment (timja) [reply plugin/config/suppressions.xml]: can't you do `maskClasses` on the maven-hpi-plugin?
or does that do something else?

can certainly exclude the snake yaml module then...

**Comment by casz** [comment]
I should move the shading out to another PR 😄

File: integrations/pom.xml [hunk 2]
```diff
@@ -15,6 +15,7 @@
     <jenkins.version>2.164.1</jenkins.version> <!-- Remove when parent base is updated. -->
     <maven.deploy.skip>true</maven.deploy.skip> <!-- no need to be deployed during release, this is a test-only module -->
     <enforcer.skip>true</enforcer.skip> <!-- no need to be enforced, this is a test-only module -->
+      <rootdir>${basedir}/..</rootdir>
```
Comment (timja) [inline integrations/pom.xml]: indenting off here

isn't there a property build into maven you can use for this anyway?

Something like?
```
maven.multiModuleProjectDirectory
```

https://stackoverflow.com/questions/3084629/finding-the-root-directory-of-a-multi-module-maven-reactor-project

Comment (casz) [reply integrations/pom.xml]: rats, didn't even see indent was off. and no maven.multiModuleProjectDirectory is not there 😢 or at least intellij says it is not.

Comment (casz) [reply integrations/pom.xml]: Never trust your IDE 😞

**Pull request merged**